purchase_invoice_plan/models/purchase.py

```diff
@@ -206,79 +206,78 @@ def action_invoice_create(self):
                 last_installment = max(installments)
                 for installment in installments:
                     # Getting invoice plan for each installment
-                    plan_lines = plan_obj.search(
+                    blines = plan_obj.search(
                         [('installment', '=', installment),
                          ('order_id', '=', order.id)])
-                    if plan_lines:
-                        for blines in plan_lines:
-                            if installment == 0:
-                                if blines.is_advance_installment:
-                                    inv_id = self._create_deposit_invoice(
-                                        blines.deposit_percent,
-                                        blines.deposit_amount,
-                                        blines.date_invoice,
-                                        blines)
-                                elif blines.is_deposit_installment:
-                                    inv_id = self._create_deposit_invoice(
-                                        blines.deposit_percent,
-                                        blines.deposit_amount,
-                                        blines.date_invoice,
-                                        blines)
+                    if blines:
+                        if installment == 0:
+                            if blines.is_advance_installment:
+                                inv_id = self._create_deposit_invoice(
+                                    blines.deposit_percent,
+                                    blines.deposit_amount,
+                                    blines.date_invoice,
+                                    blines)
+                            elif blines.is_deposit_installment:
+                                inv_id = self._create_deposit_invoice(
+                                    blines.deposit_percent,
+                                    blines.deposit_amount,
+                                    blines.date_invoice,
+                                    blines)
 
-                                blines.write({'ref_invoice_id': inv_id.id})
```

Comment:
Generally this method is just too long. Can you help make sub function, also to be more readable. Thanks.